aptos-move/framework/aptos-framework/sources/block.move

```diff
@@ -116,6 +114,31 @@ module aptos_framework::block {
         borrow_global<BlockMetadata>(@aptos_framework).height
     }
 
+    /// Emit the event and update height and global timestamp
+    fun emit_new_block_event(vm: &signer, new_block_event: NewBlockEvent) acquires BlockMetadata {
+        let block_metadata_ref = borrow_global_mut<BlockMetadata>(@aptos_framework);
+        block_metadata_ref.height = block_metadata_ref.height + 1;
+        timestamp::update_global_time(vm, new_block_event.proposer, new_block_event.time_microseconds);
+        event::emit_event<NewBlockEvent>(&mut block_metadata_ref.new_block_events, new_block_event);
```

Comment:
I am probably paranoid, but shall we expose EventHandle::count and check it matches the height here?

Reply:
no, but we can use prover to prove they always equal

Reply:
all .. that .. magic